packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -551,6 +551,71 @@ class NavigatorObserver {
 /// because it doesn't depend on its animation parameters (elided with `_`
 /// and `__` in this example). The transition is built on every frame
 /// for its duration.
+///
+/// ### Embedding Navigators
+///
+/// An app can use more than one Navigator. Embedding one Navigator below
+/// another Navigator can result in an "inner journey" such as user
+/// registration, store checkout, or other independent journeys that represent
+/// a subsection of your overall application.
+///
+/// ```dart
+/// class MyApp extends StatelessWidget {
+///  @override
+///  Widget build(BuildContext context) {
+///    return new MaterialApp(
+///      //...some parameters omitted...//
+///      // MaterialApp contains our top-level Navigator
+///      initialRoute: '/',
+///      routes: {
+///        '/': (BuildContext context) => new HomePage(),
+///        '/signup': (BuildContext context) => new SignUpPage(),
+///      },
+///    );
+///  }
+/// }
+///
+/// class SignUpPage extends StatelessWidget {
+///  @override
+///  Widget build(BuildContext context) {
+///    // SignUpPage builds its own Navigator which ends up being an embedded
+///    // Navigator in our app.
+///    return new Navigator(
+///      initialRoute: 'signup/personal_info',
+///      onGenerateRoute: (RouteSettings settings) {
+///        WidgetBuilder builder;
+///        switch (settings.name) {
+///          case 'signup/personal_info':
+///            builder = (BuildContext _) => new CollectPersonalInfoPage();
+///            break;
+///          case 'signup/choose_credentials':
+///            builder = (BuildContext _) => new ChooseCredentialsPage(
+///              onSignupComplete: () {
+///                // Referencing Navigator.of(context) from here refers to the
+///                // top level Navigator because SignUpPage is above the
+///                // embedded Navigator that it created. Therefore, this pop()
+///                // will pop the entire "sign up" journey and return to the
+///                // "/" route, AKA HomePage.
+///                Navigator.of(context).pop();
+///              },
+///            );
+///            break;
+///          default:
+///            throw new Exception('Invalid route: ${settings.name}');
+///        }
+///        return new MaterialPageRoute(builder: builder, settings: settings);
+///      },
+///    );
+///  }
+/// }
+/// ```
+///
+/// Take care to call the correct `Navigator` in your hierarchy. `Navigator.of()`
+/// finds the nearest `Navigator` ancestor from the `context` that you provide.
+/// If you define embedded `Navigator`s in one large `build()` method then you
```

Comment:
I'd just be explicit here instead of being probabilistic. e.g. if you don't retrieve a BuildContext instance from a subtree below your nested Navigator, Navigator.of will get the Navigator above it.

Reply:
I tried to formulate something like this but I'm having difficulty composing a statement that would be useful to someone who doesn't already know how this works.  Any ideas?  I know multiple contexts are a problem for developers because I've seen confusion around this before.  Until this notion of distinct "context" Objects really clicks for developers, it feels very strange and unpredictable.

Reply:
```dart
/// [Navigator.of] operates on the nearest ancestor [Navigator] from the given [BuildContext]. 
/// Be sure to provide a [BuildContext] below the intended [Navigator], especially in large 
/// [build] methods where nested [Navigator]s are created. The [Builder] widget can be used to 
/// access [BuildContext] at desired locations in the widget subtree.
```
plus line breaks.